File: 200_code_python/24_cach_1.py
# Viết bằng PYTHON
# Chuyển đổi chuỗi thành chữ hoa: Nhập một chuỗi và chuyển đổi tất cả các ký tự thành chữ hoa.

# CÁCH 1: Sử dụng bằng mã ASCII để chuyển đổi ký tự

# Nhập chuỗi từ bàn phím
s = input("Nhập một chuỗi: ")

# Chuyển chuỗi thành chữ hoa
chuoi_hoa =''
for char in s:
    # Kiểm tra nếu ký tự là chữ thường, chuyển đổi thành chữ hoa
    if 'a' <= char <= 'z:':
        chuoi_hoa += chr(ord(char) - 32) # Dùng bằng mã ASCII để chuyển đổi chữ thường thành chữ hoa
        # chr(ord(char) - 32) chuyển đổi ký tự char thành chữ hoa bằng cách trừ 32 trong bảng mã ASCII
    else:
        chuoi_hoa += char # Giữ nguyên nếu không phải chữ thường

# In chuỗi đã chuyển đổi
print(f"Chuỗi chữ hoa: {chuoi_hoa}")

File: 200_code_cpp/24_cach_1.cpp
// Viết bằng C++
// Chuyển đổi chuỗi thành chữ HOA: Nhập một chuỗi và chuyển đổi tất cả các ký tự thành chữ HOA.

// Cách 1: Sử dụng vòng lập for.
#include <iostream>
#include <string>
using namespace std;

int main() 
{
    // Nhập chuỗi từ người dùng
    string s;
    cout << "Nhap mot chuoi: ";
    getline(cin, s); // getline() để nhập chuỗi có chứa khoảng trắng

    // Chuyển đổi chuỗi thành chữ HOA
    for(int i = 0; i < s.length(); i++) // Duyệt qua từng ký tự trong chuỗi, s.length() trả về độ dài của chuỗi.
    {
        // Kiểm tra và chuyển ký tự thường thành chữ HOA

        if (s[i] >= 'a'  && s[i] <= 'z') //Nếu ký tự là chữ thường.
        {
            s[i] = s[i] - ('a' - 'A'); // Chuyển đổi ký tự thường thành HOA
        }
    }

    // In chuỗi đã chuyển đổi
    cout << "Chuoi chu HOA: " << s << endl;

    return 0;
}